sigal/plugins/nonmedia_files.py

```diff
@@ -0,0 +1,184 @@
+            if plugin_settings.get('thumb_font_color', None):
+                kwargs['font_color'] = plugin_settings['thumb_font_color']
+            if plugin_settings.get('thumb_font_size', None):
+                kwargs['font_size'] = plugin_settings['thumb_font_size']
+            generate_thumbnail(self.src_ext[1:].upper(), self.thumb_path,
+                               self.settings['thumb_size'],
+                               options=self.settings['jpg_options'],
+                               **kwargs)
+        return super().thumbnail
+
+
+def generate_thumbnail(text, outname, box,
+                       bg_color=DEFAULT_CONFIG['thumb_bg_color'],
+                       font=DEFAULT_CONFIG['thumb_font'],
+                       font_color=DEFAULT_CONFIG['thumb_font_color'],
+                       font_size=DEFAULT_CONFIG['thumb_font_size'],
+                       options=None):
+    """Create a thumbnail image."""
+
+    kwargs = {}
+    if font:
+        kwargs['font'] = ImageFont.truetype(font, font_size)
+    if font_color:
+        kwargs['fill'] = font_color
+
+    img = PILImage.new("RGB", box, bg_color)
+
+    anchor = (box[0] // 2, box[1] // 2)
+    d = ImageDraw.Draw(img)
+    logger.info(f"kwargs: {kwargs}")
+    d.text(anchor, text, anchor='mm', **kwargs)
+
+    outformat = 'JPEG'
+    logger.info('Save thumnail image: %s (%s)', outname, outformat)
+    save_image(img, outname, outformat, options=options, autoconvert=True)
+
+
+def process_nonmedia(media):
+    """Process a non-media file: copy and create thumbnail."""
+    logger.info('Processing non-media file: %s', media.dst_filename)
+    settings = media.settings
+    plugin_settings = settings.get('nonmedia_files_options', {})
+
+    try:
+        utils.copy(media.src_path, media.dst_path,
+                   symlink=settings['orig_link'])
+    except Exception:
+        if logger.getEffectiveLevel() == logging.DEBUG:
+            raise
+        else:
+            return Status.FAILURE
+
+    if plugin_settings.get('ext_as_thumb', DEFAULT_CONFIG['ext_as_thumb']):
+        logger.info('plugin_settings: %r', plugin_settings)
+        try:
+            kwargs = {}
+            if plugin_settings.get('thumb_font', None):
+                kwargs['font'] = plugin_settings['thumb_font']
+            if plugin_settings.get('thumb_bg_color', None):
+                kwargs['bg_color'] = plugin_settings['thumb_bg_color']
+            if plugin_settings.get('thumb_font_color', None):
+                kwargs['font_color'] = plugin_settings['thumb_font_color']
+            generate_thumbnail(
+                media.src_ext[1:].upper(),
+                media.thumb_path,
+                settings['thumb_size'],
+                options=settings['jpg_options'],
+                **kwargs
+            )
+        except Exception:
+            if logger.getEffectiveLevel() == logging.DEBUG:
+                raise
+            else:
+                return Status.FAILURE
```

Comment:
Would be better to return a `Status.SUCCESS` at the end (it works without but for consistency with other processors).

Reply:
Yes, you're right.  I'll change that.